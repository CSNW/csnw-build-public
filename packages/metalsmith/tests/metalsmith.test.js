const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const metalsmith = require('../')({
  layouts: './src/layouts',
  partials: './src/partials'
});
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await metalsmith.pipeline({
    cwd,
    cache: false
  });
  const files = await Files.from('**/*', { cwd, baseDir: join(cwd, 'src') });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
