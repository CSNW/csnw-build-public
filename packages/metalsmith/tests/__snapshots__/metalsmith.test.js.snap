// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures/src",
    "contents": "<!DOCTYPE html>
<head>
  <meta charset=\\"utf-8\\">
  <meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\">
  <title>A</title>
  <meta name=\\"description\\" content=\\"A page\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
</head>  <body>
Howdy A!
  </body>
</html>",
    "cwd": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures/src/a.html",
    "sourceMap": null,
  },
  "b.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures/src",
    "contents": "<!doctype html>
<html>
  <head>
    <title>B</title>
  </head>
  <body>
    Howdy B!
  </body>
</html>",
    "cwd": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/metalsmith/tests/fixtures/src/b.html",
    "sourceMap": null,
  },
}
`;
